File: java-spring-rest-mvc/src/main/java/com/spring_guru/spring_rest_mvc/controllers/BeerController.java
package com.spring_guru.spring_rest_mvc.controllers;

import com.spring_guru.spring_rest_mvc.models.Beer;
import com.spring_guru.spring_rest_mvc.services.BeerService;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Controller;

import java.util.UUID;

@Slf4j
@AllArgsConstructor
@Controller
public class BeerController {
    private final BeerService beerService;

    public Beer gerBeerById(UUID id) {
        log.debug("Get beer by id {}", id);

        return beerService.getBeerById(id);
    }
}

File: kotlin-spring-rest-mvc/src/main/kotlin/com/kotlin_spring_rest_mvc/kotlin_spring_rest_mvc/controllers/BeerController.kt
package com.kotlin_spring_rest_mvc.kotlin_spring_rest_mvc.controllers

import com.kotlin_spring_rest_mvc.kotlin_spring_rest_mvc.models.Beer
import com.kotlin_spring_rest_mvc.kotlin_spring_rest_mvc.services.BeerService
import mu.KotlinLogging
import org.springframework.http.HttpHeaders
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.*

@RestController
@RequestMapping("/api/v1/beers")
class BeerController(private val beerService: BeerService) {

    private val logger = KotlinLogging.logger {}


    @PostMapping
    fun handlePost(@RequestBody beer: Beer): ResponseEntity<Beer> {
        val savedBeer: Beer = beerService.saveNewBeer(beer)

        val headers = HttpHeaders()
        headers.add("Location", "/api/v1/beers/${savedBeer.id}")

        return ResponseEntity
            .status(HttpStatus.CREATED)
            .headers(headers)
            .body(savedBeer)
    }

    @GetMapping
    fun listBeers(): List<Beer> = beerService.listBeer()

    @GetMapping("/{id}")
    fun getBeerById(@PathVariable("id") id: UUID): Beer? {
        logger.debug { "Get beer by id $id" }
        return beerService.getBeerById(id)
    }
}